src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -101,6 +106,17 @@ private static Task ExecuteObjectResult(object? obj, HttpContext httpContext)
             }
         }
 
+        private static Task WriteToResponseAsync<T>(T? value, HttpContext httpContext, JsonTypeInfo<T> jsonTypeInfo, JsonSerializerOptions options)
+        {
+            var runtimeType = value?.GetType();
+            if (runtimeType is null || jsonTypeInfo.Type == runtimeType || jsonTypeInfo.PolymorphismOptions is not null)
```

Comment:
Nit: Should we add a null-check for `value` here as well?

Reply:
If `value` is `null` then `runtimeType` is `null` as well. I don't think we need.